File: Add Code Here/LeetCode_Sols/Easy/94. Binary Tree Inorder Traversal/Solution.java
class Solution {
    List<Integer> order = new ArrayList<Integer>();
    public List<Integer> inorderTraversal(TreeNode root) {
        traverse(root);
        return order;
    }
    
    public void traverse(TreeNode root){
        if(root == null) return;
        traverse(root.left);
        order.add(root.val);
        traverse(root.right);
    }
}

File: Add Code Here/LeetCode_Sols/Easy/94. Binary Tree Inorder Traversal/Solution.rb
class TreeNode
    attr_accessor :val, :left, :right
    def initialize(val = 0, left = nil, right = nil)
        @val = val
        @left = left
        @right = right
    end
end

def inorder_traversal(root)
    result = []
    stack = []
    current = root
    while current || stack.length > 0
        while current
            stack.push(current)
            current = current.left
        end
        current = stack.pop()
        result << current.val
        current = current.right
    end
    result
end